### PR TITLE
React-app atomic design and setcount problem

### PR DESCRIPTION
**Pull request by Ganji-NaveenKumar**
React-app for Atomic Design Structure with demo-app and  setcount problem in demo.tsx

### CODE REVIEW

File: demo-app/src/components/molecules/Loginform/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { useState } from "react";
+import TypographyComp from "../../atoms/Typograph";
+import TextFieldComp from "../../atoms/InputField";
+import ButtonOrganism from "../../organisms/ButtonOrganism";
+import { Stack } from "@mui/material";
+import "./index.css";
+const LoginForm: React.FC = () => {
+  const [email, setEmail] = useState("");
```
Comment (MajaharZemoso) [inline demo-app/src/components/molecules/Loginform/index.tsx]: add type for useState

File: demo-app/src/components/molecules/Loginform/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  const [password, setPassword] = useState("");
+  return (
+    <div id="loginform">
+      <Stack mb={5} spacing={3}>
```
Comment (MajaharZemoso) [inline demo-app/src/components/molecules/Loginform/index.tsx:27]: don't use inline styling instead you can use styled component..

File: demo-app/src/components/organisms/ButtonOrganism/index.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import ButtonComp from "../../atoms/Button";
+
+interface ButtonOrganismProps {
+  // onClick:()=> void;
```
Comment (MajaharZemoso) [inline demo-app/src/components/organisms/ButtonOrganism/index.tsx]: Remove commented lines

File: demo-app/src/components/atoms/Button/index.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import {Button} from '@mui/material';
+interface ButtonProps{
+    children:React.ReactNode;
+    style:React.CSSProperties;
+    onClick:()=>void;
+}
+const ButtonComp:React.FC<ButtonProps>=(props)=>{
```
Comment (MajaharZemoso) [inline demo-app/src/components/atoms/Button/index.tsx]: You can destructure the props here itself

eg: const ButtonComp:React.FC<ButtonProps>=(children,style,onClick)=>{

File: demo-app/src/components/atoms/InputField/index.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import {TextField} from '@mui/material';
+interface TextfieldProps{
+    label:string;
+    type:string;
+    onChanges:(event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+const TextFieldComp:React.FC<TextfieldProps>=(props)=>{
```
Comment (MajaharZemoso) [inline demo-app/src/components/atoms/InputField/index.tsx]: +1

File: demo-app/src/components/molecules/Loginform/index.tsx (hunk 1 above)
Comment (Ganji-NaveenKumar) [reply demo-app/src/components/molecules/Loginform/index.tsx]: updated

File: demo-app/src/components/molecules/Loginform/index.tsx (hunk 2 above)
Comment (Ganji-NaveenKumar) [reply demo-app/src/components/molecules/Loginform/index.tsx:27]: updated

File: demo-app/src/components/organisms/ButtonOrganism/index.tsx (hunk 3 above)
Comment (Ganji-NaveenKumar) [reply demo-app/src/components/organisms/ButtonOrganism/index.tsx]: updated

File: demo-app/src/components/atoms/Button/index.tsx (hunk 4 above)
Comment (Ganji-NaveenKumar) [reply demo-app/src/components/atoms/Button/index.tsx]: updated

File: demo-app/src/components/atoms/InputField/index.tsx (hunk 5 above)
Comment (Ganji-NaveenKumar) [reply demo-app/src/components/atoms/InputField/index.tsx]: updated

**Review by MajaharZemoso — APPROVED** [review]

**Pull request merged**